File: tests/CaseUpper.cs
namespace N {
    public partial class TTest {
        public string Foo(string val) {
            string result;
            switch (val)
            {
                case 'S': result = "one"; break;
                case 'B': result = "two"; break;
            }
            return result;
        }
    }
}

File: tests/CaseUpper.pas
namespace N;

interface

type
  TTest = public class
  public
    method Foo(val: String): String;
  end;

implementation

method TTest.Foo(val: String): String;
begin
  Case val of
    'S': result := 'one';
    'B': result := 'two';
  end;
end;

end.
